File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        vector<int> negative;
        vector<int> positive;

        for (int i=0; i<nums.size(); i++){
            if(nums[i]<0){
                negative.push_back(nums[i]);
            } else {
                positive.push_back(nums[i]);
            }
        }

        int j=0, i=0; 
        while(j<nums.size()/2){
            nums[i] = positive[j];
            nums[i+1] = negative[j];
            i+=2;
            j++;
        }

        return nums;
    }
};

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.go
func rearrangeArray(nums []int) []int {
    rearr := make([]int, len(nums))
    pos, neg := 0, 1
    for _, num := range nums {
        if num < 0 {
            rearr[neg] = num
            neg+=2
        } else {
            rearr[pos] = num
            pos+=2
        }
    }

    return rearr
}